compiler/rustc_const_eval/src/interpret/validity.rs

```diff
@@ -599,6 +605,15 @@ impl<'rt, 'mir, 'tcx: 'mir, M: Machine<'mir, 'tcx>> ValidityVisitor<'rt, 'mir, '
                 if place.layout.is_unsized() {
                     self.check_wide_ptr_meta(place.meta(), place.layout)?;
                 }
+                if self.ref_tracking.is_some()
```

Comment:
```suggestion
                // Obviously raw pointers may dangle, except when we check the final value
                // of a constant -- rustc generally assumes to not encounter dangling `AllocId`
                // in `tcx` memory.
                if self.ref_tracking.is_some()
```
Also, usually we tie the CTFE checks to `self.ctfe_mode.is_some()`, not to `ref_tracking`, so please do that here as well.